bot/cogs/bot.py

```diff
@@ -119,35 +138,59 @@ def codeblock_stripping(self, msg: str):
+                    python_syntax = any(
+                        msg.content[3:9].lower() == "python",
+                        msg.content[3:5].lower() == "py"
+                    )
+
+                    bad_ticks = msg.content[:3] in not_backticks
 
-                        log.debug(f"{msg.author} posted something that needed to be put inside python code blocks. "
-                                  "Sending the user some instructions.")
-                        howto = (f"Hey {msg.author.mention}!\n\n"
-                                 "I noticed you were trying to paste code into this channel.\n\n"
-                                 f"{codeblock_tag['tag_content']}")
+                    if bad_ticks and python_syntax:
+                        howto = (f"You are using the wrong signs, use ``` instead of"
+                                 f" {not_backticks.index(msg.content[:3])}")
 
                         howto_embed = Embed(description=howto)
-                        await msg.channel.send(embed=howto_embed)
-                        self.channel_cooldowns[msg.channel.id] = time.time()
+                    else:
+                        content = self.codeblock_stripping(msg.content)
+                        if not content:
+                            return
 
+                        # Attempts to parse the message into an AST node.
+                        # Invalid Python code will raise a SyntaxError.
+                        tree = ast.parse(content)
+
+                        # Multiple lines of single words could be interpreted as expressions.
+                        # This check is to avoid all nodes being parsed as expressions.
+                        # (e.g. words over multiple lines)
+                        if not all(isinstance(node, ast.Expr) for node in tree.body):
+                            space_left = 204
+                            # Shorten the code to 10 lines and/or 204 characters.
+                            if len(content) >= space_left:
+                                current_length = 0
+                                lines_walked = 0
+                                for line in content.splitlines(keepends=True):
+                                    if current_length+len(line) > space_left or lines_walked == 10:
+                                        break
+                                    current_length += len(line)
+                                    lines_walked += 1
+                                content = content[:current_length]+"#..."
+
+                            howto = (f"Hey {msg.author.mention}!\n"
+                                     f"I noticed you were trying to paste code into this channel.\n"
+                                     "Discord has support for Markdown, which allows you to post code with full syntax"
+                                     " highlighting. Please use these whenever you paste code, as this helps improve"
```

Comment:
always put these spaces at the end of the previous line, instead. it helps readability.